### DISCUSSION TITLE
beforeImport or beforeRun?

### DISCUSSION THREAD

**Discussion by gkiely** [vitest-dev/vitest#2010] [Ideas]
I have some code whereby I want to mock a variable before the test has imported modules.
In this case it is `window.location.pathname`.

```ts
import App from '../App'; // window.location.pathname = '/my-route'

beforeImport(() => {
  vi.stubGlobal('location', { pathname: '/my-route' });
});
```

Would evaluate as:
```ts
vi.stubGlobal('location', { pathname: '/my-route' });

import App from '../App'; // // window.location.pathname = '/my-route'
```

Looking for feedback on this idea.

**Comment by sheremet-va** [comment]
You can add `setupFile` which is imported before your test. Otherwise it's impossible, and you need to add this stub yourself at the top.

**Comment by gkiely** [comment]
For this use case I need to set it differently for each test file. I don't believe setupFiles can determine the which file is running, but even then, it would be preferable to couple it to the test file.

Ok in that case I'll set up a vite plugin.
